src/segment-loader.js

```diff
@@ -1139,20 +1139,34 @@ export default class SegmentLoader extends videojs.EventTarget {
     }
   }
 
-  handleCaptions_(simpleSegment, captions, captionStreams) {
-    // Don't need to check for abort since captions are only handled for non partial
-    // appends at the moment (therefore, they will only trigger once a segment is finished
-    // being transmuxed).
-    createCaptionsTrackIfNotExists(
-      this.inbandTextTracks_, this.hls_.tech_, captionStreams);
-    addCaptionData({
-      captionArray: captions,
-      inbandTextTracks: this.inbandTextTracks_,
-      // full segments appends already offset times in the transmuxer
-      timestampOffset: 0
+  handleCaptions_(simpleSegment, captions) {
+    if (this.checkForAbort_(simpleSegment.requestId) ||
+      this.abortRequestEarly_(simpleSegment.stats) ||
+      !captions || captions.length === 0) {
```

Comment:
Will this potentially be a case? If so, maybe a comment about how it can happen.

Reply:
👍